distributed/scheduler.py

```diff
@@ -2014,6 +2014,8 @@ def transition_no_worker_processing(self, key, stimulus_id):
                 assert ts in self.unrunnable
 
             if ws := self.decide_worker_non_rootish(ts):
+                # ^ NOTE: `ts` may actually be root-ish now, but it wasn't when it went
+                # into `no-worker`. `TaskGroup` or cluster size could have changed.
```

Comment:
As mentioned in https://github.com/dask/distributed/pull/7259/files#r1015290381 this should be impossible.